suite-common/connect-init/src/actions/prepareConnectInit.ts

```diff
@@ -0,0 +1,50 @@
+import TrezorConnect, { ConnectSettings, Manifest } from '@trezor/connect';
+import { AnyAction, createAsyncThunk } from '@reduxjs/toolkit';
+import { cardanoConnectPatch } from '../cardanoConnectPatch';
+import { lockDeviceConnectPatch } from '../lockDeviceConnectPatch';
+import { bindConnectActionsListeners, ConnectEventAction } from '../bindConnectActionsListeners';
+
+export const prepareConnectInit = createAsyncThunk<
+    void,
+    {
+        lockDevice: (payload: boolean) => AnyAction;
```

Comment:
For all action creators please use types `ActionCreatorWithPayload` or `ActionCreatorWithoutPayload` from redux-toolkit. So for example here it will be `lockDevice: ActionCreatorWithPayload<boolean>`

Reply:
I am not sure if we can use it this way because it complains about types.

![Snímek obrazovky 2022-07-13 v 14 29 22](https://user-images.githubusercontent.com/2227592/178733742-9f4c04bf-6330-48bb-94cc-3d31432dc7d9.png)

the action in **suite** package is defined as:

`export const lockDevice = (payload: boolean): SuiteAction => ({
    type: SUITE.LOCK_DEVICE,
    payload,
});
`
it takes type for SuiteAction which is defined in suite actions type:

`
export type SuiteAction =
.....
| { type: typeof SUITE.LOCK_DEVICE; payload: boolean }
....
`

Reply:
We are not not going to use type for actions, typing actions is not necessary. `SuiteAction` is useless type, because it's is not going to guard you from using action that is doing  "nothing".

Currently we are defining our actions creators somewhere and then go and add that `typeof MyActionCreator`  to some global type `SuiteAction`. But if someone decide to delete this switch case from reducer so action is not handled anymore, it's still valid because someone manually placed it to Action type. So manually defining this type gives you exactly zero guarantee that actions is handled because this `SuiteAction` type is constructed manually anyway.

In case you will notify that action is not used anymore you will manually remove action creator and type so it will throw error, but it will throw error also in case there will be only action creator without strongly typed dispatch because you can't call non-existing function that was removed.

This is little bit better with redux-toolkit `createSlice` which will generate type automatically, but it also generate action creators automatically so if you delete that case, TS will throw anyway because action creator is not created anymore.

So if you are going to always use redux-toolkit slices for generating action creators and not constructing action object manually (`{ type: "Whatever", payload... }`) we will have even better type safety that with `SuiteAction` type.

`type: "NOOP"` is special case and I would prefer to use at least `createAction` for it.

Reply:
for now solved with types:

`
type Selector<TReturnValue> = (state: any) => TReturnValue;
type SuiteCompatibleAction<TPayload = void> = (
    payload: TPayload,
) => AnyAction | ActionCreatorWithPayload<TPayload> | ActionCreatorWithoutPayload;
`